Build failure on GCC 16: -Werror=nonnull in _preview_write_image (imageio.c)

### Is there an existing issue for this?

- [x] I checked and did not find my issue in the already reported ones

### Describe the bug

darktable fails to compile on Fedora 44 (GCC 16.0.1) with `-Werror=nonnull`:

```
In function ‘_preview_write_image’,
    inlined from ‘dt_imageio_export’ at /home/andrii/Dev/darktable/src/imageio/imageio.c:1012:13:
/home/andrii/Dev/darktable/src/imageio/imageio.c:1733:3: error: argument 2 null where non-null expected [-Werror=nonnull]
 1733 |   memcpy(d->buf, in, sizeof(uint32_t) * data->width * data->height);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
cc1: all warnings being treated as errors
```

The error is in `_preview_write_image()`, which `memcpy`s from its `in` parameter without a NULL guard. GCC 16's interprocedural analysis follows function pointers and detects that `dt_imageio_export()` passes `NULL` as the `in` argument when the format mime is `"x-copy"`:

```c
// src/imageio/imageio.c:1010-1014
if(strcmp(format->mime(format_params), "x-copy") == 0)
  return (format->write_image(format_params, filename, NULL, icc_type,
                              icc_filename, NULL, 0, imgid, num, total, NULL,
                              export_masks)) != 0;
```

Since `_preview_write_image` is a `write_image` callback, GCC concludes that NULL can reach the `memcpy` and refuses to compile under `-Werror=nonnull`. The diagnostic is technically correct: `memcpy` from a NULL source is undefined behaviour even when `n` is zero.

In practice the runtime path probably never fires (the preview format never has mime `"x-copy"`), but the static analysis can't see that.

The bug has been latent since [b49bf38062](https://github.com/darktable-org/darktable/commit/b49bf38062) (Oct 2022); only newer GCC versions catch it because of stricter interprocedural NULL propagation.

### Steps to reproduce

1. Use Fedora 44 (or any distro shipping GCC 16+)
2. Clone darktable master
3. `cmake -B build && make -C build -j$(nproc)`
4. Build fails with the error above.

### Expected behavior

darktable compiles successfully with GCC 16+

### Logfile | Screenshot | Screencast

```
[ 52%] Building C object bin/CMakeFiles/lib_darktable.dir/common/ai_models.c.o
In function ‘_preview_write_image’,
    inlined from ‘dt_imageio_export’ at /home/andrii/Dev/darktable/src/imageio/imageio.c:1012:13:
/home/andrii/Dev/darktable/src/imageio/imageio.c:1733:3: error: argument 2 null where non-null expected [-Werror=nonnull]
 1733 |   memcpy(d->buf, in, sizeof(uint32_t) * data->width * data->height);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
cc1: all warnings being treated as errors
gmake[2]: *** [bin/CMakeFiles/lib_darktable.dir/build.make:2087: bin/CMakeFiles/lib_darktable.dir/imageio/imageio.c.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:3469: bin/CMakeFiles/lib_darktable.dir/all] Error 2
gmake: *** [Makefile:156: all] Error 2
```


### Commit

b49bf38062

### Where did you obtain darktable from?

self compiled

### darktable version

5.5.0+1092~g27b3bdec84

### What OS are you using?

Linux

### What is the version of your OS?

Fedora 44

### Describe your system

GCC 16.0.1 20260321 (Red Hat 16.0.1-0)

### Are you using OpenCL GPU in darktable?

None

### If yes, what is the GPU card and driver?

_No response_

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

**Suggested fix** (one line): add a NULL guard at the top of `_preview_write_image`:

```c
static int _preview_write_image(dt_imageio_module_data_t *data,
                                const char *filename,
                                const void *in,
                                ...)
{
  _imageio_preview_t *d = (_imageio_preview_t *)data;

  if(!in) return 1;            // ← added
  memcpy(d->buf, in, sizeof(uint32_t) * data->width * data->height);
  d->width = data->width;
  d->height = data->height;
  return 0;
}
```

--- a/src/imageio/imageio.c
+++ b/src/imageio/imageio.c
@@ -1730,6 +1730,7 @@ static int _preview_write_image(dt_imageio_module_data_t *data,
 {
   _imageio_preview_t *d = (_imageio_preview_t *)data;
 
+  if(!in) return 1;
   memcpy(d->buf, in, sizeof(uint32_t) * data->width * data->height);
   d->width = data->width;
   d->height = data->height;

